### PR TITLE
MAINT: nep29 2022 + testing standards + pysat RC test

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses pysat/pysat#1042 and pysat/pysat#1048

Updates to use released pysat test syntax and latest standard for NEP29 and github actions.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

via pytest

Inspection of logs at https://github.com/pysat/pysatCDAAC/actions/runs/3439721561 to confirm that pysat is only installed via the RC.

**Test Configuration**:
* Operating system: Mac OS X Monterrey
* Version number: Python 3.8.11

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

**Comment by jklenzing** [comment]
Note: may be replaced by #39

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,33 +12,33 @@ jobs:
...
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
 
+    - name: Install NEP29 dependencies
+      if: ${{ matrix.numpy_ver != 'latest'}}
+      run: |
+        pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
```
Comment (jklenzing) [inline .github/workflows/main.yml:34]:
```suggestion
        pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
        # Need to force install of pandas compliant with NEP29
        pip install "pandas<1.5"
```

**Comment by jklenzing** [comment]
Bumping this back up to get the latest testing standards in.  pyproject.toml can wait until we have a better assessment

**Comment by jklenzing** [comment]
Last pull before the RC!

**Comment by jklenzing** [comment]
Pinging all hands for a review here.  Added the workflow dispatch job too.

**Comment by jklenzing** [comment]
Pinging @aburrell and @rstoneback for a review here so I can put an RC together.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,34 +11,36 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
-        python-version: ["3.8", "3.9"]
-        numpy_ver: [latest]
+        os: ["ubuntu-latest", "windows-latest"]
```
Comment (aburrell) [inline .github/workflows/main.yml:14]: What about mac?

Comment (jklenzing) [inline .github/workflows/main.yml:14]:
```suggestion
        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
```

File: .github/workflows/pysat_rc.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+name: Test with latest pysat RC
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (jklenzing) [inline .github/workflows/pysat_rc.yml:13]:
```suggestion
        os: [ubuntu-latest, macos-latest, windows-latest]
```

File: .github/workflows/main.yml (hunk 2 above)
Comment (jklenzing) [reply .github/workflows/main.yml]: updated

**Review by aburrell — APPROVED** [review]

**Pull request merged**